### PR TITLE
Add android ID

### LINKED ISSUE #221: Supply ANDROID_ID as hardware_id
The `ANDROID_ID` is available in the android app - this should be used as the hardware id for sync purposes.

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #221 

I was hoping to just use `machine_uid::get()` and fetch from settings if that failed - but the android build wouldn't compile if `machine_uid` was included, so I've had to make it a little more specific than I'd like.

Have tested on android and the setting is updated, and makes it through to 4D:

Screen Shot 2022-07-11 at 6 28 15 PM

also ran the server locally with these changes and it still works

## Tests
- [ ] The `make android` builds successfully
- [ ] When syncing with this version, the AndroidID is sent to mSupply

### CODE REVIEW

File: server/server/Cargo.toml [hunk 1]
```diff
@@ -21,10 +21,12 @@ repository = { path = "../repository" }
 service = { path = "../service" }
 util = { path = "../util" }
 
+machine-uid = { version = "0.2.0", optional = true }
```
Comment (mark-prins) [inline server/server/Cargo.toml:24]: had to make this optional for android to build

File: server/server/src/lib.rs [hunk 2]
```diff
@@ -1,4 +1,6 @@
+#[cfg(not(target_os = "android"))]
```
Comment (mark-prins) [inline server/server/src/lib.rs:1]: not how I'd like to do it.. but the crate doesn't play well with android

File: server/server/src/lib.rs [hunk 3]
```diff
@@ -82,9 +84,19 @@ async fn run_stage0(
...
     {
+        #[cfg(not(target_os = "android"))]
+        let machine_uid = machine_uid::get().expect("Failed to query OS for hardware id");
+
+        #[cfg(target_os = "android")]
+        let machine_uid = config_settings
+            .server
+            .machine_uid
+            .clone()
+            .unwrap_or("".to_string());
```
Comment (mark-prins) [inline server/server/src/lib.rs:95]: thought not to panic if this fails 🤷

**Review by andreievg — APPROVED** [review]
Thanks Mark looks good to me

**Review by clemens-msupply — APPROVED** [review]
Looks good thanks. See comment.

File: server/server/Cargo.toml [hunk 4]
```diff
@@ -21,10 +21,12 @@ repository = { path = "../repository" }
...
+
+# OpenSSL is currently only used indirectly by external crates. For the Android build we need to use actix-cors = "0.6.1"
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply server/server/Cargo.toml:26]: odd. I've moved it back now..

**Pull request merged**